W-12206229: documentation for type parameter application (#215)
mulesoft/docs-dataweave: modules/ROOT/pages/dataweave-functions.adoc

}
----

== Type Parameters Definition

> DataWeave supports the use of type parameters in its function definitions, which are similar to generics in some programming languages. 

Reviewer: ```suggestion
DataWeave function definitions support the use of type parameters, which are similar to generics in some programming languages. 
```

> .Example Function that uses type parameters

Reviewer: ```suggestion
.Example Function that Uses Type Parameters
```

[source,dataweace,linenums]
----
%dw 2.0
output application/json

fun toArray<T>(x: T): Array<T> = [x]
---
toArray(2)
----

> DataWeave also supports adding constraints for its type parameters, which constrain the possible accepted inputs for a function.

Reviewer: ```suggestion
DataWeave also supports adding type parameter constraints, which limit accepted inputs for a function.
```

> DataWeave also supports adding constraints for its type parameters, which in turn constraints the possible accepted inputs for a function.

Reviewer: ```suggestion
DataWeave also supports adding constraints for its type parameters, which constrain the possible accepted inputs for a function.
```

> .Example Function that uses type parameters with constraints

Reviewer: ```suggestion
.Example Function that Uses Type Parameters with Constraints
```
